docs/how-to-guides/backup-rook-ceph-volumes.md

```diff
@@ -0,0 +1,235 @@
+      region   = "us-west-1"
+    }
+  }
+}
+```
+
+In the above config `region` should match the region of bucket created previously using aws CLI.
+
+Add the following content to the `lokocfg.vars` file:
+
+```tf
+velero_aws_credentials = <<EOF
+[default]
+aws_access_key_id = AWS_ACCESS_KEY_ID
+aws_secret_access_key = AWS_SECRET_ACCESS_KEY
+EOF
+```
+
+Replace the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` with appropriate credentials for the
+`velero` user.
+
+#### Deploy
+
+Execute the following command to deploy the `velero` component:
+
+```bash
+lokoctl component apply velero
+```
+
+Verify the pods in the `velero` namespace are in the `Running` state (this may take a few minutes):
+
+```console
+$ kubectl -n velero get pods
+NAME                     READY   STATUS    RESTARTS   AGE
+restic-c27rq             1/1     Running   0          2m
+velero-66d5d67b5-g54x7   1/1     Running   0          2m
+```
+
+### Step 2: Deploy sample workload
+
+If you already have an application you want to backup, then skip this step.
+
+Let us deloy a stateful application and save some demo data in it. Save the following YAML config in
+a file named `stateful.yaml`:
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: demo-ns
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  labels:
+    app: demo-app
+  name: demo-app
+  namespace: demo-ns
+spec:
+  replicas: 1
+  serviceName: "demo-app"
+  selector:
+    matchLabels:
+      app: demo-app
+  template:
+    metadata:
+      labels:
+        app: demo-app
+    spec:
+      securityContext:
+        runAsNonRoot: true
+        runAsUser: 65534
+        runAsGroup: 65534
+      containers:
+      - image: busybox:1
+        name: app
+        command: ["/bin/sh"]
+        args:
+        - -c
+        - "echo 'sleeping' && sleep infinity"
+        volumeMounts:
+        - mountPath: "/data"
+          name: data
+  volumeClaimTemplates:
+  - metadata:
+      name: data
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      resources:
+        requests:
+          storage: 100Mi
+```
+
+Execute the following command to deploy the application:
+
+```bash
+kubectl apply -f stateful.yaml
+```
+
+Verify the application is running fine:
+
+```console
+$ kubectl -n demo-ns get pods
+NAME         READY   STATUS    RESTARTS   AGE
+demo-app-0   1/1     Running   0          16s
+```
+
+Execute the following command to generate some dummy data:
+
+```console
+kubectl -n demo-ns exec -it demo-app-0 -- /bin/sh -c \
+    'dd if=/dev/zero of=/data/file.txt count=40 bs=1048576'
+```
+
+Verify that the data is generated:
+
+```console
+$ kubectl -n demo-ns exec -it demo-app-0 -- /bin/sh -c 'du -s /data'
+40960   /data
+```
+
+### Step 3: Annotate pods
+
+Annotate the pods with volumes attached to them with their volume names. Replace the pod name and
+the volume name as needed in the following command:
+
+```bash
+kubectl -n demo-ns annotate pod demo-app-0 backup.velero.io/backup-volumes=data
```

Comment:
Hm, we should explain why annotation is needed. Otherwise we can just include it in statefulset pod spec...

I guess the assumption was to highlight how to make velero also backup PVs ?

Reply:
> Hm, we should explain why annotation is needed. Otherwise we can just include it in statefulset pod spec...

Because we will need to highlight that anyway, because for a casual reader it might be very easy to miss.

> I guess the assumption was to highlight how to make velero also backup PVs ?

Yep. And this should work with any application. Like user's own app as well. The user can decide to include the annotation in their config having this step explicitly does not harm. But missing out on the step can give you a false sense of security that backup is done while in reality it is not.

Reply:
All right. Then we should add some sentence like: "We need to annotate the pod now, otherwise Velero will not back up out persistent volume. If you want your PVs to be always backed up, modify pod template in deployment spec <bla bla bla>". This will explain why the annotation is needed and will make it more comprehensive.